tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/viewers/AbstractTreeViewerTest.java

```diff
@@ -164,6 +164,144 @@ public void testExpandToLevel() {
 	}
 
 	@Test
+	public void testAutoExpandTrivialPath() {
+		TestElement modelRoot = TestElement.createModel(5, 1);
+		TestElement trivialPathRoot = modelRoot.getFirstChild();
+		fViewer.setInput(modelRoot);
+
+		fTreeViewer.setAutoExpandOnSingleChildLevels(2);
+		fTreeViewer.setExpandedStateWithAutoExpandOnSingleChild(trivialPathRoot, true);
+
+		assertTrue("The expanded widget child is not expanded", fTreeViewer.getExpandedState(trivialPathRoot));
+		assertTrue("The first child of the trivial path was not auto-expanded",
+				fTreeViewer.getExpandedState(trivialPathRoot.getFirstChild()));
+		assertFalse("Trivial path is expanded further than specified depth ",
+				fTreeViewer.getExpandedState(trivialPathRoot.getFirstChild().getFirstChild()));
+	}
+
+	@Test
+	public void testAutoExpandTrivialDeeperDownPath() {
+		TestElement modelRoot = TestElement.createModel(6, 1);
+		TestElement trivialPathRoot = modelRoot.getFirstChild();
+		fViewer.setInput(modelRoot);
+
+		fTreeViewer.setAutoExpandOnSingleChildLevels(2);
+		System.out.println("AbstractTreeViewerTest - 1" + modelRoot);
+		fTreeViewer.setExpandedStateWithAutoExpandOnSingleChild(trivialPathRoot.getFirstChild(), true);
+
+		assertFalse("The parent of the expanded widget child is expanded",
+				fTreeViewer.getExpandedState(trivialPathRoot));
+		assertTrue("The first child of the trivial path was not auto-expanded",
+				fTreeViewer.getExpandedState(trivialPathRoot.getFirstChild()));
+		assertTrue("The second child of the trivial path was not auto-expanded",
+				fTreeViewer.getExpandedState(trivialPathRoot.getFirstChild().getFirstChild()));
+		assertFalse("Trivial path is expanded further than specified depth ",
+				fTreeViewer.getExpandedState(trivialPathRoot.getFirstChild().getFirstChild().getFirstChild()));
+	}
+
+	@Test
+	public void testAutoExpandTrivialPathFromRoot() {
+		TestElement modelRoot = TestElement.createModel(5, 5);
+		TestElement trivialPathRoot = modelRoot;
+		fViewer.setInput(modelRoot);
+
+		fTreeViewer.setAutoExpandOnSingleChildLevels(2);
+		fTreeViewer.setExpandedStateWithAutoExpandOnSingleChild(trivialPathRoot, true);
+
+		assertFalse("The first child of the trivial path was auto-expanded",
+				fTreeViewer.getExpandedState(trivialPathRoot.getFirstChild()));
+	}
+
+	@Test
+	public void testAutoExpandTrivialPathFromRootWithSensibleTrivialPath() {
```

Comment:
```suggestion
	public void testAutoExpandOnSingleChildFromRootWithSensibleTrivialPath() {
```

Reply:
Addressed in https://github.com/eclipse-platform/eclipse.platform.ui/commit/fb256ed9423ce9f552aa278fa3893d833ccc0e63